### DISCUSSION TITLE
Change AppData\Local\rancher-desktop on Windows

### DISCUSSION THREAD

**Discussion by ymebrugts** [rancher-sandbox/rancher-desktop#6296] [Q&A] [answered]
We cannot use the C disk for some of our clients so I'd like to change the locations Rancher Desktop is using to another disk. For instance "D:\rancher-desktop-vmdata"

The folder with the most impact is "AppData\Local\rancher-desktop".

How can I do this? Preferably without Symlinking. 

I already looked at the issues below:
[Ability to choose VMs location #3228](https://github.com/rancher-sandbox/rancher-desktop/issues/3228)
[Selectable location for storage of VM #369](https://github.com/rancher-sandbox/rancher-desktop/issues/369)
[Where are images and vms stored? #1551](https://github.com/rancher-sandbox/rancher-desktop/discussions/1551)

**Comment by jandubois** [comment, marked as answer]
I don't have access to my Windows machine right now, so this is untested, but I expect it to work.

Rancher Desktop is using 2 WSL distros, `rancher-desktop` and `rancher-desktop-data`. The first one includes the software and is basically fixed size. The second one includes all the container images you build or pull from repositories, and can grow quite significantly.

Make sure you quit Rancher Desktop before trying to move the data. Note that the `rancher-desktop-data.tar` file will be quite large too, so put it on a disk with sufficient free space:

```console
wsl --shutdown

wsl --export rancher-desktop-data rancher-desktop-data.tar

wsl --unregister rancher-desktop-data

wsl --import rancher-desktop-data D:\rancher-desktop-vmdata rancher-desktop-data.tar --version 2
```

You can move the `rancher-desktop` volume the same way, if you need to.

Once you have verified that everything works, you can delete the `rancher-desktop-data.tar` file.

Note that doing a "factory reset" will delete both distros, so they will be recreated in their default locations, and you would have to move them again manually.

And please report back here if this actually worked! 😄

**Reply by razvan-pro** [reply]
I tried this just now and it worked for me - thanks @jandubois! 👍

**Reply by dupinderdhiman** [reply]
It helped, thank you for the solution @jandubois

**Reply by GlenHughes** [reply]
Worked for me - thank you!

**Reply by facundogoyena** [reply]
This is great, thank you! Are there any plans to introduce a setting to change this in the preferences or something similar?

**Reply by rklec** [reply]
The modern way [is described here](https://superuser.com/a/1771436/1702233) and you would use `wsl --manage <distro_name> --move <new_location>` instead.
